src/modules/hmis/ClientIdEncoder.ts

```diff
@@ -0,0 +1,64 @@
+import CryptoJS from 'crypto-js';
+import { startOfToday } from 'date-fns';
+
+export default class ClientIdEncoder {
+  static PROTECT_IDS = import.meta.env.PUBLIC_PROTECTED_IDS === 'true';
+
+  static INITIAL_DELIMITER = import.meta.env.PUBLIC_INITIAL_DELIMITER || '==';
+
+  static KEY = (
+    import.meta.env.PUBLIC_PROTECTED_ID_KEY ||
+    import.meta.env.PUBLIC_FQDN ||
+    'rX9BDyW0'
+  ).slice(0, 7);
+
+  static DELIMITER_REGEX = new RegExp(`^${this.INITIAL_DELIMITER}`);
+
+  static encode(id: number | string): string {
+    if (!this.PROTECT_IDS) return String(id);
+
+    const dayStamp = startOfToday().valueOf() / (60 * 60 * 24);
+
+    return this.obfuscate(parseInt(String(id)), dayStamp);
+  }
+
+  static obfuscate(id: number, dayStamp: number): string {
+    // TODO: Make this match the backend
```

Comment:
If this obfuscates and deobfuscates, I'm not sure why it needs to match the back-end.  It's not like the id will match, or that the paths would be the same.

Reply:
Agreed, they don't need to match